Lib/test/test_tools/test_unparse.py

```diff
@@ -263,12 +263,14 @@ def test_dict_unpacking_in_dict(self):
 
 class DirectoryTestCase(ASTTestCase):
     """Test roundtrip behaviour on all files in Lib and Lib/test."""
+    NAMES = None
 
     # test directories, relative to the root of the distribution
     test_directories = 'Lib', os.path.join('Lib', 'test')
 
-    def test_files(self):
-        # get names of files to test
+    def get_names(self):
```

Comment:
Wouldn't be better to make it a static method?

Reply:
I prefer to use a regular method, even if technically a staticmethod would work, since the method depends on the class.

Reply:
If the method depends on the class, make it a class method.

But in that case you should use `cls.NAMES` instead of `DirectoryTestCase.NAMES`.